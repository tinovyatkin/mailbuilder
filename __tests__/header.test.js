'use strict';

const {
  isAscii,
  escapeNameString,
  shouldEncodedOrEscapeAddresses,
  utf8encode,
  encodeIfNeeded,
} = require('../lib/header');

describe('header generating functions', () => {
  test('isAscii', () => {
    expect(isAscii('Is that "war"?')).toBeTruthy();
    expect(isAscii('Это война')).toBeFalsy();
    expect(isAscii('😎 - cool!')).toBeFalsy();
  });

  test('escapeNameString', () => {
    expect(escapeNameString('John Doe')).toBe('John Doe');
    expect(escapeNameString('Phil "Cool Guy" Freo')).toBe(
      '"Phil \\"Cool Guy\\" Freo"',
    );
  });

  test('shouldEncodedOrEscapeAddresses', () => {
    expect(
      shouldEncodedOrEscapeAddresses([
        { name: 'Jack Sparrow, CPA', address: 'jack@example.com' },
        { name: 'John Smith', address: 'john@example.com' },
      ]),
    ).toBeFalsy();

    expect(
      shouldEncodedOrEscapeAddresses([
        { name: 'Vasya Pupkin', address: 'vasya@pupkin.com' },
        { name: 'Petya Ivanov', address: 'петя@иванов.рф' },
      ]),
    ).toBeTruthy();

    expect(
      shouldEncodedOrEscapeAddresses([
        { name: 'Vasya Pupkin', address: 'vasya@pupkin.com' },
        { name: 'Петя Иванов', address: 'petr@ivanov.com' },
      ]),
    ).toBeTruthy();
  });

  test('utf8encode', () => {
    expect(utf8encode('😎 - cool!')).toMatchSnapshot();
    expect(utf8encode('Василий Пупкин <vas@pup.in>')).toMatchSnapshot();
  });

  test('encodeIfNeeded', () => {
    expect(encodeIfNeeded('Simple text stays as is!')).toBe(
      'Simple text stays as is!',
    );
    expect(encodeIfNeeded('😎 text gonna be encoded')).toMatch(
      /^=\?utf-8\?B\?/,
    );
  });
});
