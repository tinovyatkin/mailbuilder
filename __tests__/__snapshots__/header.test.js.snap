// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header generating functions utf8encode 1`] = `"=?utf-8?B?8J+YjiAtIGNvb2wh?="`;

exports[`header generating functions utf8encode 2`] = `"=?utf-8?B?0JLQsNGB0LjQu9C40Lkg0J/Rg9C/0LrQuNC9IDx2YXNAcHVwLmluPg==?="`;
